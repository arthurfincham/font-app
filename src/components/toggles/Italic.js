import ToggleButton from '@mui/material/ToggleButton';
import { StyledToggleButtonGroup } from '../styled/styledTheme';
import ToggleTooltip from '../ToggleTooltip';
import { ItalicContext } from '../../context/ItalicContext';
import { useContext } from 'react';
import FormatItalicIcon from '@mui/icons-material/FormatItalic';

export default function Single() {
  const { myFontItalic, setMyFontItalic } = useContext(ItalicContext);

  const handleToggle = (event, newStyle) => {
    setMyFontItalic(newStyle);
  };

  return (
    <ToggleTooltip title="Italic">
      <StyledToggleButtonGroup size="small" value={false} exclusive onChange={handleToggle} aria-label="code snippet">
        <ToggleButton value={!myFontItalic} aria-label="left aligned" data-testid="Italic-test">
          <FormatItalicIcon />
        </ToggleButton>
      </StyledToggleButtonGroup>
    </ToggleTooltip>
  );
}
