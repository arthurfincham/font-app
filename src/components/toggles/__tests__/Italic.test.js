import Italic from '../Italic';
import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import 'jest-styled-components';

Enzyme.configure({ adapter: new Adapter() });

it('it correctly renders the Italic toggle', () => {
  const wrapper = shallow(<Italic />);

  expect(toJson(wrapper)).toMatchSnapshot();
});

it('has a single button', () => {
  const wrapper = mount(<Italic />);
  expect(wrapper.find('button')).toHaveLength(1);
});
