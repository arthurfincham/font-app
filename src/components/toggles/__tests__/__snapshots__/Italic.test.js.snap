// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it correctly renders the Italic toggle 1`] = `
<ToggleTooltip
  title="Italic"
>
  <Styled(ForwardRef(ToggleButtonGroup))
    aria-label="code snippet"
    exclusive={true}
    onChange={[Function]}
    size="small"
    value={false}
  >
    <ForwardRef(ToggleButton)
      aria-label="left aligned"
      data-testid="Italic-test"
      value={true}
    >
      <Memo(ForwardRef(FormatItalicIcon)) />
    </ForwardRef(ToggleButton)>
  </Styled(ForwardRef(ToggleButtonGroup))>
</ToggleTooltip>
`;
